test/test/protojure/pedestal_test.clj

```diff
@@ -43,14 +43,17 @@
   {:status   200
    :headers  {"Content-Type" "application/text"}
    :trailers {"Meta" "test"}
-   :body     {:key "clojure is awesome"}})
+   :body     "{:key \"clojure is awesome\"}"})
```

Comment:
since this is get-edn, we should probably either a) make that true, or b) change the name to get-text or something like that.

re (a), the proper way is probably to drop the Content-type header and let the body-params interceptor convert the map -> [application/edn, (pr ..)].  IIRC, this body-params interceptor will get triggered when there is no content type and the body is a map

Reply:
I did not follow the body-params interceptor comment -- I'm pretty sure that interceptor is purely for parsing bodies coming in on a request, rather than handling server responses [But I did not actually go and find the pedestal ns that does handle writing sequences out in the jetty implementation].

This test now passes with both application/text or application/edn. Strangely, it does not pass without a content type set (its not immediately obvious to me why, or if its a an issue on the cli-http client end with defaulting to html with no header set etc.), I have not dug into why we don't see the behavior implemented in: https://github.com/pedestal/pedestal/blob/d20065013abf5d3793ae5301e18a2398707fa2a9/service/src/io/pedestal/http/impl/servlet_interceptor.clj, but I assume it has to do with our use of a custom chain-provider.

Reply:
I've returned the default to the normal case by just special casing the IFn dispatch type with transmit body -- while the test still fails if the `"Content-Type" "application/edn"` header is not set, this maintains previous behavior.

Reply:
What I was referring to I think is represented here under "Clojure collection":

http://pedestal.io/reference/response-bodies

I may have mis-attributed that to the body-params interceptor rather than the servlet.  To that point, that probably explains the issue you mention in the second paragraph, because I assume the servlet handling of the EDN case also adds the application/edn header, whereas our handler currently does not.  It probably is wise to retain that Clojure collection behavior of the servlet in our handler, I would imagine.  I do not know how this impacts what issue you are addressing, though.

Reply:
Yes, to be clear, I did preserve this behavior by creating an additional defmethod dispatching on IPersistentCollection for the edn case, and IFn for the json-body case.

I think that addresses the concern? We now support returning edn, but only when the header is provided by the returning handler (we fully support the json-body interceptor behavior, because that co-locates the json transform and the header setting).

Reply:
The issue this PR solves is that right now, if a collection is returned and the json-body interceptor (added to the tests in this PR) is present, a string version of the function returned by the json-body interceptor is sent over the wire.

This modifies that behavior to now call the function with an output stream as an argument to the function, and then write out the string corresponding to that output stream.

This version of the PR no longer modifies the edn behavior in away, was the PR was updated to only modify the behavior of transmit-body on the IFn case, rather than the initial PR version which changed the :default behavior.

Reply:
Ah, I had encountered this a few months back, and I interpreted it as some kind of problem with the pedestal json interceptor, and switched to the ring json interceptor instead (which pre-renders to a string and thus would not suffer from the issue you are highlighting).  What you describe makes sense.  Let me review in that light.